Junimatic/Patcher.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using HarmonyLib;
+using StardewValley;
+
+using static NermNermNerm.Stardew.LocalizeFromSource.SdvLocalize;
+
+namespace NermNermNerm.Junimatic
+{
+    internal static class Patcher
+    {
+        /// <summary>
+        /// Apply the harmony patches for Junimatic
+        /// </summary>
+        internal static void Apply()
+        {
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Farmer), nameof(Farmer.addItemToInventoryBool)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Farmer_addItemToInventoryBool_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createItemDebris)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createItemDebris_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createObjectDebris), [typeof(string), typeof(int), typeof(int), typeof(int), typeof(int), typeof(float), typeof(GameLocation)]),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createObjectDebris_Prefix))
+            );
+        }
+
+        internal static List<Item> InterceptHarvest(Action harvestAction, GameLocation machineLocation)
+        {
+            var playerField = typeof(Game1).GetField(I("_player"), System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
```

Comment:
What were the reasons behind going this route instead of overriding get_player? Also good call on overwriting _player, not Game1.Player as I was warned explicitly not to do that due to the getter calling unload.

Reply:
This isn't calling the setter for the property, 'Player', which indeed unloads the farmer.  Note it's setting the backing field directly.  I think it's preferable to patching Game1.Player because that property gets called *very* frequently.  Adding a patch adds overhead.  Not much overhead, but in a thing that gets called that often, it can add up.

Reply:
Ok. Per when I was asking about this in the discord awhile back, setting _player wasn't discouraged. In fact, it was what I was going to do until I realized what the code I was referencing from Garden Pot Automate was doing (intercepting the call to Game1.Player and passing something else instead).